src/datatypes.ts

```diff
@@ -193,71 +187,120 @@ export function serializeTime(value: Time): string {
     }
   }
 
+  if (value.timezoneHourOffset || value.timezoneHourOffset == 0) {
+    const timezoneFormatted =
+      Math.abs(value.timezoneHourOffset) < 10
+        ? "0" + Math.abs(value.timezoneHourOffset)
+        : Math.abs(value.timezoneHourOffset);
+
+    timezoneString =
+      value.timezoneHourOffset >= 0
+        ? "+" + timezoneFormatted
+        : "-" + timezoneFormatted;
+
+    if (value.timezoneMinuteOffset) {
+      timezoneString =
+        timezoneString +
+        ":" +
+        (value.timezoneMinuteOffset < 10
+          ? "0" + value.timezoneMinuteOffset
+          : value.timezoneMinuteOffset);
+    } else {
+      timezoneString = timezoneString + ":00";
+    }
+  }
+
   return (
-    (utcHours < 10 ? "0" + utcHours : utcHours) +
+    (value.hour < 10 ? "0" + value.hour : value.hour) +
     ":" +
-    (utcMinutes < 10 ? "0" + utcMinutes : utcMinutes) +
+    (value.minute < 10 ? "0" + value.minute : value.minute) +
     ":" +
     (value.second < 10 ? "0" + value.second : value.second) +
-    (value.millisecond ? "." + millisecondString + "Z" : "Z")
+    (value.millisecond ? "." + millisecondString : "") +
+    (timezoneString ? timezoneString : "")
   );
 }
 
 /**
  * @internal
  * @param literalString Value to deserialise.
  * @returns Deserialized time, or null if the given value is not a valid serialised datetime.
- * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-repr
  */
 export function deserializeTime(literalString: string): Time | null {
   // Time in the format described at
-  // https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
-  // (without constraints on the value).
+  // https://www.w3.org/TR/xmlschema-2/#time-lexical-repr
   // \d\d:\d\d:\d\d - Two digits for the hour, minute and second, respectively, separated by a `:`.
   //                  Example: "13:37:42".
   // (\.\d+)? - Optionally a `.` followed by one or more digits representing milliseconds.
   //            Example: ".1337".
   // (Z|(\+|-)\d\d:\d\d) - The letter Z indicating UTC, or a `+` or `-` followed by two digits for
   //                       the hour offset and two for the minute offset, separated by a `:`.
   //                       Example: "+13:37".
-  const datetimeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)/;
+  const datetimeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)?/;
```

Comment:
Heh... Super minor issue, but I guess this is a `timeRegEx` only:

```suggestion
  const timeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)?/;
```